### PR TITLE
Krmahadevan reduce complexity

### PR DESCRIPTION
**Pull request by krmahadevan**
Changeset involves the following 

* Reduce cognitive complexity in TestNG class.
* Reduce cognitive complexity in Assert class.

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/TestNG.java [hunk 1]
```diff
@@ -1226,60 +1236,63 @@ private void generateReports(List<ISuite> suiteRunners) {
    * until an alternative mechanism is found.
    */
   public List<ISuite> runSuitesLocally() {
+    if (m_suites.isEmpty()) {
+      setStatus(HAS_NO_TEST);
+      error("No test suite found. Nothing to run");
+      usage();
+      return Lists.newArrayList();
```
Comment (juherr) [inline src/main/java/org/testng/TestNG.java]: `Collections.emptyList()`?

File: src/main/java/org/testng/TestNG.java [hunk 2]
```diff
@@ -1226,60 +1236,63 @@ private void generateReports(List<ISuite> suiteRunners) {
...
-          Thread.currentThread().interrupt();
-          error("Error waiting for concurrent executors to finish " + handled.getMessage());
-        }
-      }
-    }
-    else {
-      setStatus(HAS_NO_TEST);
-      error("No test suite found. Nothing to run");
-      usage();
+      return Lists.newArrayList(suiteRunnerMap.values());
```
Comment (juherr) [inline src/main/java/org/testng/TestNG.java]: `Collections.unmodifiableList(...)`?

File: src/main/java/org/testng/Assert.java [hunk 3]
```diff
@@ -853,16 +850,16 @@ public static void assertEquals(Iterator<?> actual, Iterator<?> expected) {
    * @param message the assertion error message
    */
   public static void assertEquals(Iterator<?> actual, Iterator<?> expected, String message) {
-    if(actual == expected) {
+    if (actual == null && expected == null) {
```
[the remark this answers is not in the document]
Comment (krmahadevan) [reply src/main/java/org/testng/Assert.java]: I can't remember why I changed it. So I will revert it back.

File: src/main/java/org/testng/TestNG.java (hunk 2 above)
Comment (krmahadevan) [reply src/main/java/org/testng/TestNG.java]: `suiteRunnerMap.values()` returns a `Collection<ISuite>` and `Collections.unmodifiableList()` takes in a `List` and not a `Collection`. So the only choice I have would be to pass the `Lists.newArrayList()` call to `Collections.unmodifiableList()`. Is that required ?

File: src/main/java/org/testng/TestNG.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/TestNG.java]: fixed.

File: src/main/java/org/testng/TestNG.java (hunk 2 above)
Comment (juherr) [reply src/main/java/org/testng/TestNG.java]: Ok, forget it :)

**Review by juherr — APPROVED** [review]
LGTM! 👍 
Ping @cbeust

**Comment by krmahadevan** [comment]
ping @cbeust - Can you please help me with this as well ?

**Pull request merged**